src/panel/poi/PoiPanel.jsx

```diff
@@ -178,9 +180,32 @@ export default class PoiPanel extends React.Component {
   }
 
   backToList = () => {
+    const { poiFilters } = this.props;
+
     Telemetry.add(Telemetry.POI_BACKTOLIST);
     fire('restore_location');
-    window.app.navigateTo(`/places/?type=${this.props.sourceCategory}`);
+
+    let uri = '/places/';
+
+    // Reconstruct uri
+    for (const filterName of Object.keys(poiFilters)) {
```

Comment:
That would be cleaner and easier to manage in a pure function, like `getFiltersAsQueryString`, which would return a string like `type=Foo&q=Bar`.
Or maybe better, adding a generic function in `url_utils` to add a set of query string parameters to any url, and just  renaming the keys in this file.

Reply:
That's the perfect use case for `URLSearchParams`, which can be initialized with an object:
```js
const params = new URLSearchParams({type: 'restaurant' , q: 'McDo'});
console.log(params.toString()); // "type=restaurant&q=McDo"
```

Reply:
yep, but I had issues doing this, will retry